src/index.ts

```diff
@@ -1,2 +1,3 @@
-export { jsonToSchema } from "./schema-generator";
+export { jsonToSchema, addRule } from "./schema-generator";
+export * from "./schema-generator/types";
 import "./custom-rules";
```

Comment:
I might have missed this previously. Why is this `import`?

Reply:
there are a bunch of custom validation rules that need to be applied to Yup, these individual rules are automatically applied as soon as the files are ran, so all that's needed is to import those files. alternatively I can write in a different way where we import each rule and apply them individually elsewhere, will it be clearer that way?

Reply:
yea that might be clearer

Reply:
have updated to use named exports instead, there is a new `applyCustomRules()` to apply all the rules

however the call to `applyCustomRules()` is still in the main index file because it appears to be the most suitable place to use. I considered importing in the `schema-generator/` folder but that looks out of place and there is a circular dependency issue too.